### DISCUSSION TITLE
Battery Indicator

### DISCUSSION THREAD

**Discussion by electricalhog** [TG-Techie/TG-Watch-Software#20] [Ideas]
# The Idea
Ideally, I would like to have a graphical battery icon that would change to represent both the battery level and its charging state. 
## Method 1
Using a few custom icons that would be switched out based on the variables could create the nicest looking results as the exact appearance could be tweaked to make it look the best. The downsides here are probably speed, design time, and making it stateful.
## Method 2
This could probably also be done with a progress bar with a stateful palette. The progress bar would need to be modified to advance from right to left with the colored portion. I think this is a very doable method if this idea ever gets worked on, however, it will require some tg_gui rework to make colors work properly (no offense, but the color implementation isn't great). It won't look amazing, but still very good for what it is.

**Comment by TG-Techie** [comment]
This would be a great idea to implement.  

## Method 1
This sounds like the best way to go about adding it, especially if the charge indicator is a lightning bolt/etc.  "design time, and making it stateful" both sound like challenges that will take some time to figure out, great point.

## Method 2
😂 no offense taken, the color "system" is a hack. I've been reticent to implement a proper theme/style/color system as cross-platform color control varies from OS to OS when using native widgets. It may just be better to say "eff it" and implement one that works well for the watch and accept there will be breaking changes in the future. Thoughts? 
(it could all be `_private` like label alignment is for now)
